### DISCUSSION TITLE
Find a node by attribute value

### DISCUSSION THREAD

**Discussion by iamursky** [ueberdosis/tiptap#1946] [General Questions] [answered]
I'm trying to find a position of heading that has id attribute with some value to apply `scrollIntoView` to it. How do I do it?

**Comment by KaneCohen** [comment, marked as answer]
Answered in discord, but for future reference if it might have somebody looking for similar issue, here's a possible solution (that I didn't test):

```ts
editor.state.doc.forEach(node, offset) => {
  if (node.attrs.id === yourId) {
    editor.chain().setNodeSelection(offset).scrollIntoView().run();
  }
});
```

**Comment by Priestch** [comment]
You can set some attribute of node DOM when render HTML, then just query the DOM, and use the scrollIntoView method.

**Reply by iamursky** [reply]
Yes, that worked until I've changed the layout. Seems that at some moment I did something wrong with flexboxes.
